src/main/java/org/inferred/freebuilder/processor/Analyser.java

```diff
@@ -166,10 +170,88 @@ Metadata analyse(TypeElement type) throws CannotGenerateCodeException {
       metadataBuilder
           .clearProperties()
           .addAllProperties(codeGenerators(properties, baseMetadata, builder.get()));
+
+      metadataBuilder.addAllSuperBuilderTypes(superBuilders(type));
+      metadataBuilder.putAllSuperTypeProperties(
+              processSuperTypeProperties(type, baseMetadata, builder));
     }
     return metadataBuilder.build();
   }
 
+  private ImmutableMap<ParameterizedType, ImmutableList<Property>> processSuperTypeProperties(
+          TypeElement type,
+          Metadata baseMetadata,
+          Optional<TypeElement> builder) throws CannotGenerateCodeException {
+    Map<ParameterizedType, ImmutableList<Property>> toRet =
+            new HashMap<ParameterizedType, ImmutableList<Property>>();
+
+    final ImmutableSet<TypeElement> superTypes = MethodFinder.getSupertypes(type);
+    for (TypeElement superType : superTypes) {
+      if (superType.equals(type)) {
+        continue;
+      }
+
+      final ImmutableSet<ExecutableElement> superMethods = methodsOn(superType, elements);
+      final Map<ExecutableElement, Property> superPropertiesRet =
+              findProperties(superType, superMethods);
+      if (superPropertiesRet.isEmpty()) {
+        continue;
+      }
+
+      ParameterizedType pType = QualifiedName.of(superType).withParameters(
+              superType.getTypeParameters());
+
+      // Code builder dance
+      if (builder.isPresent()) {
+        final Metadata metadataSuperType = analyse(superType);
+        final Metadata.Builder metadataBld = Metadata.Builder.from(metadataSuperType);
+        metadataBld.setBuilderFactory(Optional.<BuilderFactory>absent());
+
+        for (Map.Entry<ExecutableElement, Property> entry : superPropertiesRet.entrySet()) {
+          Config config = new ConfigImpl(
+                  builder.get(),
+                  metadataBld.build(),
+                  entry.getValue(),
+                  entry.getKey(),
+                  ImmutableSet.<String>of());
+
+          entry.setValue(new Property.Builder()
+                  .mergeFrom(entry.getValue())
+                  .setCodeGenerator(createCodeGenerator(config))
+                  .build());
+        }
+      }
+
+      toRet.put(pType, ImmutableList.copyOf(superPropertiesRet.values()));
+    }
+
+    return ImmutableMap.copyOf(toRet);
+  }
+
+  private ImmutableSet<ParameterizedType> superBuilders(TypeElement type)
+          throws CannotGenerateCodeException {
+    Set<ParameterizedType> toRet = new HashSet<ParameterizedType>();
+    PackageElement pkg = elements.getPackageOf(type);
+
+    final ImmutableSet<TypeElement> superTypes = MethodFinder.getSupertypes(type);
+    for (TypeElement superType : superTypes) {
+      if (superType.equals(type)) {
+        continue;
+      }
+
+      final Optional<AnnotationMirror> freeBuilderMirror =
+              findAnnotationMirror(superType, FreeBuilder.class);
```

Comment:
This won't work: the FreeBuilder annotation is source-only. If your two types are not compiled together, or if your IDE decides to compile the two classes separately (e.g. you change your subtype in Eclipse), you'll be given the class definition from the compiled JAR. Check out the logic in BuildablePropertyFactory, for example. In this case, however, you'll need to check for the presence of all the getter methods too, since a user might have generated a broken builder by hand.